src/main/kotlin/org/pkl/lsp/documentation/DocCommentMemberLinkProcessor.kt

```diff
@@ -0,0 +1,129 @@
+    )?
+    """
+        .trimIndent()
+    )
+
+  private val keywords = setOf("null", "true", "false", "this", "unknown", "nothing")
+
+  fun process(docComments: String, ownerNode: PklDocCommentOwner): String {
+    return docComments.replace(linkRegex) { match ->
+      val (memberLink, customLinkText) =
+        if (match.groups[2] == null) {
+          match.groups[1]!!.value to null
+        } else {
+          match.groups[2]!!.value to match.groups[1]!!.value
+        }
+      val linkText = customLinkText ?: memberLink
+      if (memberLink in keywords) {
+        "`$linkText`"
+      } else {
+        when (val resolved = resolveDocLink(memberLink, ownerNode)) {
+          is PklNode -> {
+            buildString {
+              append("[`$linkText`](")
+              append(resolved.getLocationUri(forDocs = true))
+              append(")")
+            }
+          }
+          else -> {
+            "`$memberLink`"
+          }
+        }
+      }
+    }
+  }
+
+  private fun resolveDocLink(text: String, ownerNode: PklNode): PklNode? {
+    var currentNode: PklNode? = ownerNode
+    val parts = text.split('.')
+    if (parts.isEmpty()) return null
+    val first = parts[0]
+    currentNode =
+      when {
+        first.endsWith("()") -> currentNode?.resolveMethod(first.dropLast(2))
+        else -> currentNode?.resolveVariable(first) ?: return null
+      }
+    for (i in 1..parts.lastIndex) {
+      val part = parts[i]
+      currentNode =
+        when {
+          part.endsWith("()") -> currentNode?.resolveMethod(part.dropLast(2)) ?: return null
+          else -> currentNode?.resolveVariable(part) ?: return null
+        }
+    }
+    return currentNode
+  }
+
+  private fun PklNode.resolveMethod(methodName: String): PklNode? {
+    val context = containingFile.pklProject
+    return when (val element = this) {
+      is PklClass ->
+        element.cache(context).methods[methodName] ?: element.parent?.resolveMethod(methodName)
+      is PklModule -> element.cache(context).methods[methodName]
+      else -> element.parent?.resolveMethod(methodName)
+    }
+  }
+
+  private fun PklNode.resolveVariable(variableName: String): PklNode? {
+    val context = containingFile.pklProject
+    return when (val element = this) {
+      is PklClass ->
+        element.typeParameterList?.resolveVariable(variableName)
+          ?: element.cache(context).properties[variableName]
+          ?: element.parent?.resolveVariable(variableName)
+      is PklModule ->
+        when {
+          variableName == "module" -> this
+          else -> {
+            element.imports
+              .find { it.memberName == variableName }
+              ?.let { import ->
+                if (import.isGlob) import else import.resolveModules(context).first()
+              }
+              ?: element.cache(context).typeDefsAndProperties[variableName]
+              ?: project.pklBaseModule.module.cache(null).typeDefsAndProperties[variableName]
```

Comment:
Why not pass `context` here to the cache?

Reply:
Providing context here will give us different cached members per context, which doesn't make sense for the base module because they are always the same.